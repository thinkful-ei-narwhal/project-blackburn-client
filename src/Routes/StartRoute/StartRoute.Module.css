.story-list {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.start-page-form {
  display: inline-block;
}

.story_panel {
  border: solid 1pt black;
  width: 200px;
  margin: 10px;
}
