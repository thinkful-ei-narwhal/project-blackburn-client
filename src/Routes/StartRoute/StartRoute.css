.story_panel {
    /* border: 1px solid gray; */
    box-shadow: 4px 4px 4px 4px lightgray;
    margin: 10px;
    padding: 5px;
    width: 25vh;
    height: 100%;
    justify-self: center;


    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    justify-content: center;
    align-items: center;
}
.story_panel:hover {
    border: 1px solid blue;
}
.inputform {
    width: 5vh;
    height: 5vh;
}

.story_panel:hover {
    box-shadow: 8px 8px 8px 8px lightgray;
    flex-grow: 3;
}

.select-difficulty {
    display: inline-flex;
    flex-direction: column;
    justify-content: flex-start;
    margin: 10vh 3vh 3vh 3vh;
    padding: 1vh;
}

.story-list {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
}


.start-header {
    text-align: center;
}


.back-btn {
    margin-top: 10px;
    margin-right: 10px;
    text-decoration: none;
    border: 1px solid #D9C1AA;
    padding: 15px;
    color: #D9C1AA;
    border-radius: 6px;
    background-color: black;
}

.back-btn:hover {
    background-color: #AE9172;
    opacity: 75%;
    color: #D9C1AA
}