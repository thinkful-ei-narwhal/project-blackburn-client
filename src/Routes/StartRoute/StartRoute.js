import React, { Component } from 'react';
import Button from '../../Components/Button/Button';
import storyService from '../../Services/story-api-service';
import BlackburnContext from '../../Context/BlackburnContext';
import { Link, Redirect } from 'react-router-dom';
import { Input, Label } from './../../Components/Form/Form';
import './StartRoute.css';

class StartRoute extends Component {
  static contextType = BlackburnContext;
  constructor(props) {
    super(props);
    this.state = {
      story_id: null,
      difficulty_setting: 'medium',
      stories: [],
      redirect: false,
    };
    this.handleStorySubmit = this.handleStorySubmit.bind(this);
  }

  componentDidMount() {
    storyService.getAllStories().then((stories) => {
      this.setState({ stories: stories });
    });
  }

  handleDifficulty = (e) => {
    this.setState({ difficulty_setting: e.target.value });
  };

  handleInputSelect = (e) => {
    this.setState({ story_id: e.target.value });
  };

  handleStorySubmit = (e) => {
    e.preventDefault();
    this.context.setStoryState(
      this.state.story_id,
      this.state.difficulty_setting
    );
    this.setState({ redirect: true });
  };

  renderStories() {
    return (
      //youre figuring out how to select a radio button
      //change img on line 37 to back ground of line 36 if we wish
      <div className="story-list">
        {this.state.stories.map((story) => (
          <label key={story.id} className="story-label" htmlFor={`${story.id}`}>
            <div className="story_panel" id={`story_panel ${story.id}`}>
              <img src={story.story_thumbnail} width="75px" alt="Story Art" />
              <h2>{story.story_name}</h2>
              <p>{story.story_synopsis}</p>
              <input
                className="inputform"
                type="radio"
                name="story_select"
                id={story.story_name}
                value={story.id}
                onChange={this.handleInputSelect}
                selected={this.state.story_id === `${story.id}`}
              />
            </div>
          </label>
        ))}
      </div>
    );
  }
  render() {
    return (
      <div className="start-container">
        {this.state.redirect ? (
          <Redirect to={'/storypage'} />
        ) : (
          <>
            <Link to="/dashboard" className="dashboard-lnk">
              <Button className="back-btn"> &#8592; </Button>
            </Link>
            <h1 className="start-header">Select a Story</h1>
            <div className="startpage-main">
              <form
                className="start-page-form"
                id="story_form"
                onSubmit={(e) => this.handleStorySubmit(e)}
              >
                <label className="difficulty-label" htmlFor="select_difficulty">
                  Difficulty
                </label>
                <select
                  className="select-difficulty"
                  id="select_difficulty"
                  defaultValue="medium"
                  onChange={this.handleDifficulty}
                >
                  <option value="easy">Easy</option>
                  <option value="medium">Medium</option>
                  <option value="hard">Hard</option>
                </select>
                <Button
                  disabled={
                    !!(
                      this.state.story_id === null ||
                      this.state.difficulty_setting === null
                    )
                  }
                  className="difficulty-btn"
                  type="submit"
                >
                  Start
                </Button>
                {this.renderStories()}
              </form>
            </div>
          </>
        )}
      </div>
    );
  }
}
export default StartRoute;
