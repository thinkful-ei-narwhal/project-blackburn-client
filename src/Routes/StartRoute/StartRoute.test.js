import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import BlackBurnContext from '../../Context/BlackburnContext';
import StartRoute from './StartRoute';

describe(`Renders start page to dom without crashing`, () => {
    it('renders checkpoint win crashing', () => {
        const wrapper = shallow(
        <BlackBurnContext.Provider>
            <StartRoute />
        </BlackBurnContext.Provider>
    )
    .find('.start-container')
    expect(toJson(wrapper)).toMatchSnapshot()
    })
})