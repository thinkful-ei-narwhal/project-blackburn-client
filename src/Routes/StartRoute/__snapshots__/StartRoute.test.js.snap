// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders start page to dom without crashing renders checkpoint win crashing 1`] = `null`;
